Various rewordings, punctuation additions, and light fixes to the Developer Guide. (#16967)
nvaccess/nvda: projectDocs/dev/developerGuide/developerGuide.md

The NVDA Add-on API includes all NVDA internals, except symbols that are prefixed with an underscore.

> The NVDA Add-on API changes over time, such as by the addition of new features, removal or replacement of outdated libraries, deprication of unused or replaced code and methodologies, and changes to Python.

Author: ```suggestion
The NVDA Add-on API changes over time, for example because of the addition of new features, removal or replacement of outdated libraries, deprecation of unused or replaced code and methodologies, and changes to Python.
```

### Gestures {#TranslatingGestures}

The gestures defined originally in NVDA are configured to expect English software and keyboard layout.
In most cases, these gestures can also be executed on other keyboard layouts without any problem.
However, sometimes a gesture originally defined by NVDA is not adapted for a specific locale (keyboard layout or software).
The need to modify an original gesture may be due to the following reasons:

* The original gesture is defined with a character that is not a key name on the locale keyboard layout.
Generally, the key names are the characters that can be input without the help of a modifier key (`shift`, `control`, etc.)
* The original gesture takes advantage of the keys physical location on the English keyboard layout, but this advantage does not exist with the locale keyboard layout.
* The original gesture is defined to match a native shortcut in Windows or in an application, but the shortcut in the local version of Windows or of this application is not the same as the English one.

> In all of these cases, NVDA allows to remap this gesture for this specific locale.

Author: ```suggestion
In all of these cases, NVDA allows remapping of this gesture for this specific locale.
```

### Parsing additional command line arguments in your plugin {#PluginCLIArgs}

By default NVDA accepts a limited set of command line arguments and shows an error for unknown ones.
However if you want to use any additional arguments, this is possible by adding a handler to the [extension point](#ExtensionPoints) `addonHandler.isCLIParamKnown`.
Note that since command line arguments are processed just after NVDA starts, your add-on needs to process them in a global plugin, since app modules or other drivers may not be loaded at this stage.
A sample handler can be written as follows:

```py
def processArgs(cliArgument: str) -> bool:
	if cliArgument == "--enable-addon-feature":
		# Code to process your argument...
		return True  # Argument is known to the add-on and should not be flagged by NVDA
	return False  # unknown argument - NVDA should warn user
```

> Then the handler needs to be registered--preferably in the constructor of your global plugin:

Reviewer: ```suggestion
Then the handler needs to be registered, preferably in the constructor of your global plugin:
```
